sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -123,7 +123,7 @@ class DataFrame protected[sql](
     })
   }
 
-  @transient protected[sql] val logicalPlan: LogicalPlan = queryExecution.logical match {
+  @transient protected[sql] lazy val logicalPlan: LogicalPlan = queryExecution.logical match {
```

Comment:
should be lazy, because command like explain does not need the logical plan to be executed.